course_UvA-DL/activation-functions/Activation_Functions.py

```diff
@@ -0,0 +1,824 @@
+        super().__init__()
+
+        # Create the network based on the specified hidden sizes
+        layers = []
+        layer_sizes = [input_size] + hidden_sizes
+        for layer_index in range(1, len(layer_sizes)):
+            layers += [nn.Linear(layer_sizes[layer_index - 1], layer_sizes[layer_index]), act_fn]
+        layers += [nn.Linear(layer_sizes[-1], num_classes)]
+        # nn.Sequential summarizes a list of modules into a single module, applying them in sequence
+        self.layers = nn.Sequential(*layers)
+
+        # We store all hyperparameters in a dictionary for saving and loading of the model
+        self.config = {
+            "act_fn": act_fn.config,
+            "input_size": input_size,
+            "num_classes": num_classes,
+            "hidden_sizes": hidden_sizes
+        }
+
+    def forward(self, x):
+        x = x.view(x.size(0), -1)  # Reshape images to a flat vector
+        out = self.layers(x)
+        return out
+
+
+# %% [markdown]
+# We also add functions for loading and saving the model.
+# The hyperparameters are stored in a configuration file (simple json file):
+
+
+# %%
+def _get_config_file(model_path, model_name):
+    # Name of the file for storing hyperparameter details
+    return os.path.join(model_path, model_name + ".config")
+
+
+def _get_model_file(model_path, model_name):
+    # Name of the file for storing network parameters
+    return os.path.join(model_path, model_name + ".tar")
+
+
+def load_model(model_path, model_name, net=None):
+    """Loads a saved model from disk.
+
+    Inputs:
+        model_path - Path of the checkpoint directory
+        model_name - Name of the model (str)
+        net - (Optional) If given, the state dict is loaded into this model. Otherwise, a new model is created.
+    """
+    config_file, model_file = _get_config_file(model_path,
+                                               model_name), _get_model_file(model_path, model_name)
+    assert os.path.isfile(
+        config_file
+    ), f"Could not find the config file \"{config_file}\". Are you sure this is the correct path and you have your model config stored here?"
+    assert os.path.isfile(
+        model_file
+    ), f"Could not find the model file \"{model_file}\". Are you sure this is the correct path and you have your model stored here?"
+    with open(config_file) as f:
+        config_dict = json.load(f)
+    if net is None:
+        act_fn_name = config_dict["act_fn"].pop("name").lower()
+        act_fn = act_fn_by_name[act_fn_name](**config_dict.pop("act_fn"))
+        net = BaseNetwork(act_fn=act_fn, **config_dict)
+    net.load_state_dict(torch.load(model_file, map_location=device))
+    return net
+
+
+def save_model(model, model_path, model_name):
+    """Given a model, we save the state_dict and hyperparameters.
+
+    Inputs:
+        model - Network object to save parameters from
+        model_path - Path of the checkpoint directory
+        model_name - Name of the model (str)
```

Comment:
```suggestion
    Args:
        model: Network object to save parameters from
        model_path: Path of the checkpoint directory
        model_name: Name of the model (str)
```
I think this is the correct notation